tests/pos-macros/i6997c.scala

```diff
@@ -6,9 +6,9 @@ inline def mcr(x: => Any): Any = ${mcrImpl('x)}
 
 def mcrImpl(body: Expr[Any])(using ctx: QuoteContext): Expr[Any] =
   body match
-    case '{$x: $t} =>
+    case '{$x: $T} =>
       '{
-        val tmp: $t = $x
+        val tmp: T = $x
```

Comment:
This is not as symmetric as before. I think meta-programmers will be at a loss about whether to use `$` or not before a type.

Reply:
It is not symmetric because it is not intended to be as quoted types do not follow the same level symmetry that expression do. The syntax `$` for types won't exist outside of patterns and hence they will have a single option and won't be able to get confused about this at the use site. The use of `$` in the pattern is its own issue (see #10050) which we will address separately. Here we focus on the use sites and keep the changes to the patterns to a minimum.